macrosynergy/pnl/historic_portfolio_volatility.py

```diff
@@ -129,15 +129,20 @@ def _estimate_variance_covariance(
 
     cov_matr = np.zeros((len(piv_ret.columns), len(piv_ret.columns)))
 
+    nan_ratio = lambda x: np.isnan(x).sum() + (x == 0).sum() / len(x)
+    apply_mask = lambda x: x if nan_ratio(x) <= nan_tolerance else np.nan * x
+
     for iB, cB in enumerate(piv_ret.columns):
         for iA, cA in enumerate(piv_ret.columns[: iB + 1]):
+            rA: np.ndarray = apply_mask(piv_ret[cA].values)
```

Comment:
Sorry for chase you on this one again @Magnus167, but [PEP8](https://peps.python.org/pep-0008/) requires lower case as the Python convention.

Reply:
Also, we need to apply the mask consistently for each, so they keep the same length between the two vectors `rA` and `rB` - especially as we drop the index when we make them into numpy arrays (`.values`).